samcli/commands/sync/command.py

```diff
@@ -373,31 +374,49 @@ def do_cli(
                                 dependency_layer,
                             )
                         else:
-                            execute_infra_contexts(build_context, package_context, deploy_context)
+                            code_sync_resources = execute_infra_contexts(build_context, package_context, deploy_context)
+
+                            if code_sync_resources:
+                                resource_ids = [str(resource) for resource in code_sync_resources]
+
+                                LOG.info("The following resources will be code synced for an update: %s", resource_ids)
+                                execute_code_sync(
+                                    template_file,
+                                    build_context,
+                                    deploy_context,
+                                    sync_context,
+                                    resource_ids,  # type: ignore
```

Comment:
Why we need to ignore type here?

Reply:
Because click returns a type of Tuple[str], that's what we defined in the execute_code_sync, and resource_ids is a list as defined above.

Reply:
In that case, we should convert into the type that we are willing to use instead of ignoring the type checking (which might lead into problems in the future).

Reply:
We have sufficient unit tests to prevent such risks, I really don't see the benefit of having an additional operation here for the type conversion